### PR TITLE
Misc Improvements

### PR DESCRIPTION
**Pull request by Alex-Programs**
Changed welcome screen, began throttle display support, limited camera zoom-in, flare update, weapon name assignments, improved search.

### CODE REVIEW

File: src/viewer/app.ts [hunk 1]
```diff
@@ -92,6 +92,8 @@ class Application {
 	public sceneManager = new SceneManager(this);
 	private mapLoader = new MapLoader(this.sceneManager);
 
+	isOfflineTest = false;
```
Comment (Strikeeaglechase) [inline src/viewer/app.ts]: Make sure variables have been marked as public, private, or protected

File: src/viewer/entities/playerVehicle.ts [hunk 2]
```diff
@@ -59,6 +63,12 @@ class PlayerVehicle extends Entity {
 			const speed = addCommas(Math.floor(msToKnots(this.velocity.length())));
 			this.textOverlay.edit(`${this.owner.pilotName} [${this.displayName}]\n${Math.floor(mToFt(this.position.y))}ft\n${speed}kn`);
 		}
+
+		if (this.app.isOfflineTest && this.lastTestingCMSDispense + 6000 < Date.now()) {
```
Comment (Strikeeaglechase) [inline src/viewer/entities/playerVehicle.ts]: Testing code like this should go into the offlineTestSetup, you can use `setInterval` to create a loop that calls the fireCm method at some rate.

File: src/viewer/entityBase/entity.ts [hunk 3]
```diff
@@ -76,6 +76,8 @@ class Entity {
 	public iMeshId: number;
 	private iMeshOffsetObject: THREE.Object3D;
 
+	private engineOffsets: object | null;
```
Comment (Strikeeaglechase) [inline src/viewer/entityBase/entity.ts]: Don't use `object` typing, its equivalent to `any`, this should be `Vector[]`

File: src/viewer/meshLoader/meshLoader.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ interface EntityMeshConfig {
 	scaleDamper?: number;
 	loadScale?: number;
 	color?: string;
+	engineOffsets: object | null;
```
Comment (Strikeeaglechase) [inline src/viewer/meshLoader/meshLoader.ts]: Don't use `object`. Should be an optional type so we don't have to put an empty array on every config object. The type should probably be `engineOffsets?: Vector[]`

File: src/viewer/meshLoader/meshLoader.ts [hunk 5]
```diff
@@ -282,6 +292,13 @@ class MeshLoader {
 
 		return config.scaleDamper;
 	}
+
+	public getEngineOffsets(entityKey: string) {
+		const config = entityMeshs.find(em => em.key == entityKey);
+		if (!config || !config.scaleDamper) return null;
```
Comment (Strikeeaglechase) [inline src/viewer/meshLoader/meshLoader.ts]: Does `scaleDamper` affect engine offsets?

**Review by Strikeeaglechase — CHANGES_REQUESTED** [review]

**Comment by Alex-Programs** [comment]
poke

**Review by Strikeeaglechase — APPROVED** [review]
pog ty

**Pull request merged**